libmcount/wrap.c

```diff
@@ -300,7 +305,7 @@ __visible_default int backtrace(void **buffer, int sz)
 	if (!check_thread_data(mtdp))
 		mcount_rstack_restore(mtdp);
 
-	pr_dbg("%s is called from [%d]\n", __func__, mtdp->idx);
+	pr_dbg("%s is called from [%d]\n", __func__, get_mtdp_idx(mtdp));
```

Comment:
I found the real crash at `mtdp->idx` access here, but we can also put it above like this.
```c
	if (!check_thread_data(mtdp)) {
		mcount_rstack_restore(mtdp);
		pr_dbg("%s is called from [%d]\n", __func__, mtdp->idx);
	}
```